cmd/kube-proxy/app/server_test.go

```diff
@@ -142,72 +142,97 @@ func Test_detectNodeIPs(t *testing.T) {
 	cases := []struct {
 		name           string
 		rawNodeIPs     []net.IP
-		bindAddress    string
+		nodeIPOverride []string
+		ipFamilyPolicy v1.IPFamilyPolicy
 		expectedFamily v1.IPFamily
 		expectedIPv4   string
 		expectedIPv6   string
 	}{
 		{
-			name:           "Bind address IPv4 unicast address and no Node object",
+			name:           "NodeIPOverride IPv4 unicast address and no Node object",
 			rawNodeIPs:     nil,
-			bindAddress:    "10.0.0.1",
+			nodeIPOverride: []string{"10.0.0.1"},
+			ipFamilyPolicy: v1.IPFamilyPolicyPreferDualStack,
 			expectedFamily: v1.IPv4Protocol,
 			expectedIPv4:   "10.0.0.1",
 			expectedIPv6:   "::1",
 		},
 		{
-			name:           "Bind address IPv6 unicast address and no Node object",
+			name:           "NodeIPOverride IPv4 unicast address, no Node object and SingleStack IPFamilyPolicy",
 			rawNodeIPs:     nil,
-			bindAddress:    "fd00:4321::2",
+			nodeIPOverride: []string{"10.0.0.1"},
+			ipFamilyPolicy: v1.IPFamilyPolicySingleStack,
+			expectedFamily: v1.IPv4Protocol,
+			expectedIPv4:   "10.0.0.1",
+		},
+		{
+			name:           "NodeIPOverride IPv6 unicast address and no Node object",
+			rawNodeIPs:     nil,
+			nodeIPOverride: []string{"fd00:4321::2"},
+			ipFamilyPolicy: v1.IPFamilyPolicyPreferDualStack,
 			expectedFamily: v1.IPv6Protocol,
 			expectedIPv4:   "127.0.0.1",
 			expectedIPv6:   "fd00:4321::2",
 		},
 		{
-			name:           "No Valid IP found and no bind address",
+			name:           "NodeIPOverride IPv6 unicast address, no Node object and SingleStack IPFamilyPolicy",
 			rawNodeIPs:     nil,
-			bindAddress:    "",
+			nodeIPOverride: []string{"fd00:4321::2"},
+			ipFamilyPolicy: v1.IPFamilyPolicySingleStack,
+			expectedFamily: v1.IPv6Protocol,
+			expectedIPv6:   "fd00:4321::2",
+		},
+		{
+			name:           "No Valid IP found and no NodeIPOverride",
+			rawNodeIPs:     nil,
+			nodeIPOverride: []string{""},
```

Comment:
`nodeIPOverride` should never be `[]string{""}`. If you don't specify it it would be `nil`.